tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrenctyTestCase.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.config.spring;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import org.mule.api.MuleException;
+import org.mule.api.lifecycle.Disposable;
+import org.mule.api.lifecycle.Startable;
+import org.mule.tck.junit4.FunctionalTestCase;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+public class SpringRegistryConcurrenctyTestCase extends FunctionalTestCase
+{
+
+    private static final int THREAD_COUNT = 10;
+    private static final long LATCH_TIMEOUT = 5;
+    private static final TimeUnit LATCH_TIME_UNIT = TimeUnit.SECONDS;
+
+    private CountDownLatch latch;
+    private List<Thread> threads;
+    private List<Map<String, Object>> reads;
+
+    @Override
+    protected String[] getConfigFiles()
+    {
+        return new String[] {};
+    }
+
+    @Override
+    protected void doSetUp() throws Exception
+    {
+        latch = new CountDownLatch(THREAD_COUNT);
+        threads = new ArrayList<>(THREAD_COUNT);
+        reads = new ArrayList<>(THREAD_COUNT);
+    }
+
+    @Test
+    public void concurrentReads() throws Exception
+    {
+        final int registrySize = muleContext.getRegistry().lookupByType(Object.class).values().size();
+
+        for (int i = 0; i < THREAD_COUNT; i++)
```

Comment:
This kind of test could be flaky. What is needed is to test that one thread can read while another one is already reading and that when one thread is writing then any other thread will be blocked
https://github.com/mulesoft/mule/pull/1668/files#